components/board/genericlinux/board.go

```diff
@@ -66,47 +81,49 @@ func newBoard(
 		b.gpios[fmt.Sprintf("%d", pinNumber)] = b.createGpioPin(mapping)
 	}
 
-	if err := b.Reconfigure(ctx, nil, conf); err != nil {
+	if err := b.ReconfigureParsedConfig(ctx, conf); err != nil {
 		return nil, err
 	}
 	return &b, nil
 }
 
-func (b *sysfsBoard) Reconfigure(
+// Reconfigure reconfigures the board with interrupt pins, spi and i2c, and analogs.
+func (b *SysfsBoard) Reconfigure(
 	ctx context.Context,
 	_ resource.Dependencies,
 	conf resource.Config,
 ) error {
-	b.mu.Lock()
-	defer b.mu.Unlock()
-
 	newConf, err := resource.NativeConfig[*Config](conf)
 	if err != nil {
 		return err
 	}
 
-	if err := b.reconfigureSpis(newConf); err != nil {
+	return b.ReconfigureParsedConfig(ctx, newConf)
+}
+
+// ReconfigureParsedConfig is a helper for Reconfigure.
```

Comment:
This comment leaves out the most important part. If it was merely a helper, it would be named with a lower-case r so it would be private within this struct. It's broken out separately and made public because it's also used by the customlinux boards. 

...frankly, I wonder if the customlinux implementation can just call `Reconfigure` directly, without needing to break this out separately. Why didn't we do that? (I suspect the answer is "we needed a `resource.Config` but all we had was a `genericlinux.Config`, and I didn't know how to turn one into the other." I don't know how to turn one into the other, either, but I'm confident there's some way to do it, and I think that would be the preferred approach.

Reply:
there's a config clash between the generic linux config struct and the custom Linux config struct so this function has to be exported such that the `newconf.SPIs` can be accessed.